vpr/src/route/connection_router.cpp

```diff
@@ -589,26 +580,19 @@ void ConnectionRouter<Heap>::timing_driven_add_to_heap(const t_conn_cost_params&
         //
         //Pre-heap prune to keep the heap small, by not putting paths which are known to be
         //sub-optimal (at this point in time) into the heap.
-        t_heap* next_ptr = heap_.alloc();
 
+        update_cheapest(next, to_node);
         // Use the already created next path structure pointer when RCV is enabled
-        if (rcv_path_manager.is_enabled()) rcv_path_manager.move(next_ptr->path_data, next.path_data);
-
-        //Record how we reached this node
-        next_ptr->cost = next.cost;
-        next_ptr->R_upstream = next.R_upstream;
-        next_ptr->backward_path_cost = next.backward_path_cost;
-        next_ptr->index = to_node;
-        next_ptr->set_prev_edge(from_edge);
-
-        if (rcv_path_manager.is_enabled() && current->path_data) {
-            next_ptr->path_data->path_rr = current->path_data->path_rr;
-            next_ptr->path_data->edge = current->path_data->edge;
-            next_ptr->path_data->path_rr.emplace_back(from_node);
-            next_ptr->path_data->edge.emplace_back(from_edge);
+        if (rcv_path_manager.is_enabled()) {
```

Comment:
Would it be cleaner to move the code in the if statement inside update_cheapest?